projects/packages/sync/src/modules/class-terms.php

```diff
@@ -273,12 +273,16 @@ public function filter_blacklisted_taxonomies( $args ) {
 	}
 
 	/**
-	 * Filter out set_object_terms actions where the terms have not changed.
+	 * Filter out set_object_terms actions with blacklisted taxonomies or where the terms have not changed.
 	 *
 	 * @param array $args Hook args.
-	 * @return array|boolean False if no change in terms, the original hook args otherwise.
+	 * @return array|boolean False if blacklisted taxonomy or no change in terms, the original hook args otherwise.
 	 */
 	public function filter_set_object_terms_no_update( $args ) {
+		// Check if the taxonomy is blacklisted. $args[3] is the taxonomy.
+		if ( in_array( $args[3], Settings::get_setting( 'taxonomies_blacklist' ), true ) ) {
```

Comment:
```suggestion
		if ( isset( $args[3] ) && in_array( $args[3], Settings::get_setting( 'taxonomies_blacklist' ), true ) ) {
```

Just in case `$args` is not the expected array. 

Also maybe it's worth memoizing ` Settings::get_setting( 'taxonomies_blacklist' )` so that we only query the DB once under the context of a single request?

Reply:
Thanks for the suggestion, I commited it!

> Also maybe it's worth memoizing  Settings::get_setting( 'taxonomies_blacklist' ) so that we only query the DB once under the context of a single request?

Settings depend on an option, no? I feel like we don't need to memoize them, or do we?

Reply:
Mostly an optimization to avoid querying the corresponding option (DB/Cache) multiple times in a single request. Not a blocker tho!